packages/apps/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -947,18 +928,9 @@ export class JobService {
     paymentEntity.jobId = jobEntity.id;
     paymentEntity.source = PaymentSource.BALANCE;
     paymentEntity.type = PaymentType.WITHDRAWAL;
-    if (
-      (dto instanceof JobFortuneDto || dto instanceof JobCvatDto) &&
-      dto.currency === JobCurrency.USD
-    ) {
-      paymentEntity.amount = -usdTotalAmount;
-      paymentEntity.currency = JobCurrency.USD;
-      paymentEntity.rate = 1;
-    } else {
-      paymentEntity.amount = -tokenTotalAmount;
-      paymentEntity.currency = TokenId.HMT;
-      paymentEntity.rate = div(1, rate);
-    }
+    paymentEntity.amount = -totalPaymentAmount; // In the currency used for the payment.
+    paymentEntity.currency = dto.paymentCurrency;
+    paymentEntity.rate = paymentCurrencyRate;
```

Comment:
I think it makes more sense to store an actual rate conversion (for example usd -> hmt (escrow fund token))